src/ts/src/channel-log.ts

```diff
@@ -32,44 +33,103 @@ import { AlogCoreSingleton } from './core';
 export class ChannelLog {
 
   public channel: string;
-  private coreInstance: any;
+  private coreInstance: AlogCoreSingleton;
 
   constructor(channel: string) {
     this.channel = channel;
-    this.coreInstance = AlogCoreSingleton.getInstance() as any;
+    this.coreInstance = AlogCoreSingleton.getInstance();
   }
 
   /*-- Log Functions --*/
 
-  public fatal(...args: any[]): void {
-    this.coreInstance.fatal(this.channel, ...args);
+  public fatal(
+    channel: string,
+    argThree: string|MessageGenerator|LogMetadata,
+    argFour?: string|MessageGenerator|LogMetadata,
+    argFive?: LogMetadata,
+  ): void {
```

Comment:
Good news! You can define overloads for method definitions. Given that `argThree` (etc) will appear in editors, this is a user interface, and this one isn't particularly helpful. 


Also, `channel`? in a `ChannelLog` is that real? I removed it from my suggestion b/c I think it's there by accident. Also there's nothing about log codes which I see in comments, I'm guessing at their order 😅 

```suggestion
  public fatal(log: string|MessageGenerator);
  public fatal(log: string|MessageGenerator, meta: LogMetadata);
  public fatal(log: string|MessageGenerator, code?: number, meta?: LogMetadata, ): void {
```

Also we should probably create a Type for "loggable" which is just `string|MessageGenerator` to make typing easier?

Reply:
**Note** I added some details (above) about how you could use types to save yourself some.. typing 😏